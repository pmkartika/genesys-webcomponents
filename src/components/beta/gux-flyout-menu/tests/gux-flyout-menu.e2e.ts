import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-flyout-menu', () => {
  const html = `
    <gux-flyout-menu-beta>
      <span slot="target">Example Target Element</span>
      <gux-menu slot="menu">
        <gux-menu-option>Option One</gux-menu-option>
        <gux-submenu label="Submenu Two">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-submenu label="Submenu Two">
            <gux-menu-option>Option One</gux-menu-option>
            <gux-menu-option>Option Two</gux-menu-option>
            <gux-menu-option>Option Three</gux-menu-option>
          </gux-submenu>
          <gux-menu-option>Option Three</gux-menu-option>
        </gux-submenu>
        <gux-menu-option>Option Three</gux-menu-option>
        <gux-menu-option>Option Four</gux-menu-option>
        <gux-submenu label="Submenu Five">
          <gux-menu-option>Option One</gux-menu-option>
          <gux-menu-option>Option Two</gux-menu-option>
          <gux-submenu label="Submenu Three">
            <gux-menu-option>Option One</gux-menu-option>
            <gux-submenu label="Submenu Two">
              <gux-menu-option>Option One</gux-menu-option>
              <gux-submenu label="Submenu Two">
                <gux-menu-option>Option One</gux-menu-option>
                <gux-menu-option>Option Two</gux-menu-option>
                <gux-menu-option>Option Three</gux-menu-option>
              </gux-submenu>
              <gux-menu-option>Option Three</gux-menu-option>
            </gux-submenu>
            <gux-menu-option>Option Three</gux-menu-option>
          </gux-submenu>
        </gux-submenu>
      </gux-menu>
    </gux-flyout-menu-beta>
  `;

  describe('#render', () => {
    it(`should render as expected`, async () => {
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-flyout-menu-beta');
      await a11yCheck(page);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('hover', () => {
    it('opens flyout menu', async () => {
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-flyout-menu-beta');
      const menuWrapper = await element.find('pierce/.gux-flyout-menu-wrapper');

      expect(menuWrapper).not.toHaveClass('gux-shown');
      await element.hover();
      await page.waitForChanges();
      await a11yCheck(page);

      expect(menuWrapper).toHaveClass('gux-shown');
    });
    it('opens submenus', async () => {
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-flyout-menu-beta');
      await element.hover();
      await page.waitForChanges();
      const submenu = await page.find('gux-submenu');
      const submenuWrapper = await submenu.find('pierce/.gux-submenu-wrapper');
      await submenu.hover();
      await page.waitForChanges();
      await a11yCheck(page);

      expect(submenuWrapper).toHaveClass('gux-shown');
    });
  });
});
