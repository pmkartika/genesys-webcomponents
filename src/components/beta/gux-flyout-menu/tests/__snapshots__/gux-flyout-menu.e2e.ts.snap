// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-flyout-menu #render should render as expected 1`] = `"<gux-flyout-menu-beta tabindex=\\"0\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"hydrated\\"> <span slot=\\"target\\">Example Target Element</span> <gux-menu slot=\\"menu\\" role=\\"menu\\" class=\\"hydrated\\"><!----> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Four</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Five\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Five</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Three\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Three</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-submenu label=\\"Submenu Two\\" class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-submenu-button\\" role=\\"menuitem\\" tabindex=\\"-1\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\"><span class=\\"gux-submenu-button-text\\">Submenu Two</span><gux-icon class=\\"gux-submenu-open-icon hydrated\\" icon-name=\\"chevron-small-right\\"></gux-icon></button><div class=\\"gux-submenu-wrapper\\" data-popper-reference-hidden data-popper-escaped data-popper-placement=\\"right-start\\" style=\\"position: absolute; inset: 0px auto auto 0px; margin: 0px; transform: translate(0px, 0px);\\"><div role=\\"menu\\" class=\\"gux-submenu-content\\"> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option One</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Two</span></button></gux-menu-option> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> <gux-menu-option class=\\"hydrated\\"><!----><button type=\\"button\\" class=\\"gux-menu-option-button\\" role=\\"menuitem\\" aria-haspopup=\\"false\\" tabindex=\\"-1\\"><span class=\\"gux-menu-option-button-text\\">Option Three</span></button></gux-menu-option> </div></div></gux-submenu> </div></div></gux-submenu> </gux-menu> </gux-flyout-menu-beta>"`;
